core/corehttp/gateway_handler.go

```diff
@@ -36,6 +37,16 @@ const (
 
 var onlyAscii = regexp.MustCompile("[[:^ascii:]]")
 
+// HTML-based redirect for errors which can be recovered from, but we want
+// to provide hint to people that they should fix things on their end.
+var redirectTemplate = template.Must(template.New("redirect").Parse(`<!DOCTYPE html><html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><meta http-equiv="refresh" content="10;url={{.RedirectURL}}" /><link rel="canonical" href="{{.RedirectURL}}" /></head><body><pre>{{.ErrorMsg}}</pre><pre>(if a redirect does not happen in 10 seconds, use "{{.SuggestedPath}}" instead)</pre></body></html>`))
```

Comment:
Some newlines would be nice (allowed by backtick strings).

Reply:
Fixed in https://github.com/ipfs/go-ipfs/pull/7930/commits/b81b7549d34e0b549c0035286bdd02b2be50030c